apps/lockfile-explorer-web/src/parsing/lfxGraphLoader.ts

```diff
@@ -0,0 +1,363 @@
+    }
+
+    result.packageJsonFolderPath = packageJsonFolderPath.toString();
+    result.entryId = 'project:' + result.packageJsonFolderPath;
+    result.entryPackageName = packageName.toString();
+    if (duplicates?.has(result.entryPackageName)) {
+      const fullPath = new Path(rawEntryId).makeAbsolute('/').toString().substring(1);
+      result.displayText = `Project: ${result.entryPackageName} (${fullPath})`;
+      result.entryPackageName = `${result.entryPackageName} (${fullPath})`;
+    } else {
+      result.displayText = 'Project: ' + result.entryPackageName;
+    }
+  } else {
+    result.displayText = rawEntryId;
+
+    const match = packageEntryIdRegex.exec(rawEntryId);
+
+    if (match) {
+      const [, packageName, versionPart] = match;
+      result.entryPackageName = packageName;
+
+      const underscoreIndex = versionPart.indexOf('_');
+      if (underscoreIndex >= 0) {
+        const version = versionPart.substring(0, underscoreIndex);
+        const suffix = versionPart.substring(underscoreIndex + 1);
+
+        result.entryPackageVersion = version;
+        result.entrySuffix = suffix;
+
+        //       /@rushstack/eslint-config/3.0.1_eslint@8.21.0+typescript@4.7.4
+        // -->   @rushstack/eslint-config 3.0.1 (eslint@8.21.0+typescript@4.7.4)
+        result.displayText = packageName + ' ' + version + ' (' + suffix + ')';
+      } else {
+        result.entryPackageVersion = versionPart;
+
+        //       /@rushstack/eslint-config/3.0.1
+        // -->   @rushstack/eslint-config 3.0.1
+        result.displayText = packageName + ' ' + versionPart;
+      }
+    }
+
+    // Example:
+    //   common/temp/default/node_modules/.pnpm
+    //     /@babel+register@7.17.7_@babel+core@7.17.12
+    //     /node_modules/@babel/register
+    result.packageJsonFolderPath =
+      `common/temp/${subspaceName}/node_modules/.pnpm/` +
+      result.entryPackageName.replace('/', '+') +
+      '@' +
+      result.entryPackageVersion +
+      (result.entrySuffix ? `_${result.entrySuffix}` : '') +
+      '/node_modules/' +
+      result.entryPackageName;
+  }
+
+  parseDependencies(result.dependencies, result, rawYamlData);
+
+  return result;
+}
+
+/**
+ * Transform any newer lockfile formats to the following format:
+ * [packageName]:
+ *     specifier: ...
+ *     version: ...
+ */
+function getImporterValue(
+  importerValue: ILockfileImporterV5 | ILockfileImporterV6,
+  pnpmLockfileVersion: PnpmLockfileVersion
+): ILockfileImporterV5 {
+  if (pnpmLockfileVersion === PnpmLockfileVersion.V6) {
+    const v6ImporterValue = importerValue as ILockfileImporterV6;
+    const v5ImporterValue: ILockfileImporterV5 = {
+      specifiers: {},
+      dependencies: {},
+      devDependencies: {}
+    };
+    for (const [depName, depDetails] of Object.entries(v6ImporterValue.dependencies || {})) {
+      v5ImporterValue.specifiers![depName] = depDetails.specifier;
+      v5ImporterValue.dependencies![depName] = depDetails.version;
```

Comment:
Why are these non-null assertions required?

You could wait to construct the `v5ImporterValue` until the end to avoid these.

Reply:
Downgrading V6 \-\> V5 is not the right way to implement compatibility. This code will get deleted soon, so not worth fixing.